lrtable/src/lib/statetable.rs

```diff
@@ -127,7 +127,11 @@ where
                 .unwrap(),
             false
         );
-        let mut gotos = HashMap::with_hasher(BuildHasherDefault::<FnvHasher>::default());
+        let max = usize::from(grm.rules_len()) * usize::from(sg.all_states_len());
```

Comment:
I think we need an assert along the lines of `assert!(max < StIdx::max_value())` just in case someone manages to create a grammar with `StIdx::max_value()` items.

Reply:
Ah, yes, good idea! That was my concern that someone creates a grammar with exactly `u16::max_value()` items in which case nothing would fail and it just silently would create weird behavior.

Reply:
Okay, changed this to `max < usize::from(StIdx::max_value())`, but for some reason the assertion fails with the Java7 grammar.

Reply:
Seems like the java7 grammar has 1148 states and 241 rules (1148*241=276668), which means we are way over the 65k limit of a `u16`.

Reply:
Ah! The assertion is bollocks, obviously I only need to check `assert!(sg.all_states_len() < StIdx::max_value())`!

Reply:
Aha, yes, I agree. [Sorry, I didn't engage my thinking brain in my first comment!]